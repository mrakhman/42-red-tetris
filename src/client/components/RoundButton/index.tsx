import { TouchableOpacity, View, Text, ViewStyle } from 'react-native';
import React from 'react';

const RoundButton = ({
  size, color, label, style, onPress, disabled, text,
}: {
  size: number, color: string, label?: string,
  style?: ViewStyle, onPress?: () => void, disabled?: boolean, text?: string,
}): JSX.Element => {
  const button = React.useRef<TouchableOpacity>(null);

  return (
    <View style={[{ alignItems: 'center' }, style]}>
      <TouchableOpacity
        ref={button}
        disabled={disabled}
        onPress={() => onPress?.()}
        style={{
          backgroundColor: color,
          borderRadius: size / 2,
          height: size,
          width: size,
          shadowColor: 'rgba(0,0,0, .4)',
          shadowOffset: { height: 1, width: 1 },
          shadowOpacity: 20,
          shadowRadius: 5,
          justifyContent: 'center',
          alignItems: 'center',
        }} >
        {text && <Text style={{ fontSize: 25 }}>{text}</Text>}
      </TouchableOpacity>
      {label && <Text style={{ marginTop: 10 }}>{label}</Text>}
    </View>
  );
};

export default RoundButton;
