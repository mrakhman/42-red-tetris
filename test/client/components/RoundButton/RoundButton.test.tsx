import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import RoundButton from '/client/components/RoundButton';

describe('Snapshots', () => {
  it('should render RoundButton with all props', () => {
    const props = {
      size: 50,
      color: 'red',
      label: 'Button',
      style: { marginTop: 20 },
      onPress: jest.fn(),
      disabled: false,
      text: 'ðŸŽ¶',
    };

    const wrapper = shallow(<RoundButton {...props} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should render RoundButton with just necessary props', () => {
    const props = {
      size: 50,
      color: 'red',
      label: undefined,
      style: undefined,
      onPress: undefined,
      disabled: undefined,
      text: undefined,
    };

    const wrapper = shallow(<RoundButton {...props} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
