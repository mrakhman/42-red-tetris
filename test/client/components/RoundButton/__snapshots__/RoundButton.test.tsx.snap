// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshots should render RoundButton with all props 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
      },
      Object {
        "marginTop": 20,
      },
    ]
  }
>
  <ForwardRef
    disabled={false}
    onPress={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "red",
        "borderRadius": 25,
        "height": 50,
        "justifyContent": "center",
        "shadowColor": "rgba(0,0,0, .4)",
        "shadowOffset": Object {
          "height": 1,
          "width": 1,
        },
        "shadowOpacity": 20,
        "shadowRadius": 5,
        "width": 50,
      }
    }
  >
    <Text
      style={
        Object {
          "fontSize": 25,
        }
      }
    >
      ðŸŽ¶
    </Text>
  </ForwardRef>
  <Text
    style={
      Object {
        "marginTop": 10,
      }
    }
  >
    Button
  </Text>
</View>
`;

exports[`Snapshots should render RoundButton with just necessary props 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
      },
      undefined,
    ]
  }
>
  <ForwardRef
    onPress={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "red",
        "borderRadius": 25,
        "height": 50,
        "justifyContent": "center",
        "shadowColor": "rgba(0,0,0, .4)",
        "shadowOffset": Object {
          "height": 1,
          "width": 1,
        },
        "shadowOpacity": 20,
        "shadowRadius": 5,
        "width": 50,
      }
    }
  />
</View>
`;
